worlds/oot_soh/location_access/dungeons/forest_temple.py

```diff
@@ -2,8 +2,465 @@
+    ])
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_LOBBY, world, [
+        (Regions.FOREST_TEMPLE_SOUTH_CORRIDOR, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_NORTH_CORRIDOR, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, lambda bundle: (can_use(Items.SONG_OF_TIME, bundle) or is_child(bundle))),
+        (Regions.FOREST_TEMPLE_NE_OUTDOORS_LOWER, lambda bundle: (can_use(Items.FAIRY_BOW, bundle) or can_use(Items.FAIRY_SLINGSHOT, bundle))),
+        (Regions.FOREST_TEMPLE_WEST_CORRIDOR, lambda bundle: (small_keys(Items.FOREST_TEMPLE_SMALL_KEY, 1, bundle))),
+        (Regions.FOREST_TEMPLE_EAST_CORRIDOR, lambda bundle: False),
+        (Regions.FOREST_TEMPLE_BOSS_REGION, lambda bundle: has_item(LocalEvents.DEFEATED_MEG, bundle)),
+        (Regions.FOREST_TEMPLE_BOSS_ENTRYWAY, lambda bundle: False)
+    ])
+    
+    ## Forest Temple North Corridor
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_NORTH_CORRIDOR, world, [
+        (Regions.FOREST_TEMPLE_LOBBY, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_LOWER_STALFOS, lambda bundle: True)
+    ])
+    
+    ## Forest Temple Lower Stalfos
+    # Events
+    add_events(Regions.FOREST_TEMPLE_LOWER_STALFOS, world, [
+        (EventLocations.FOREST_TEMPLE_LOWER_STALFOS_FAIRY_POT, LocalEvents.LOWER_STALFOS_FAIRY, lambda bundle: True),
+    ])
+    # Locations
+    add_locations(Regions.FOREST_TEMPLE_LOWER_STALFOS, world, [
+        (Locations.FOREST_TEMPLE_FIRST_STALFOS_CHEST, lambda bundle: (can_kill_enemy(bundle, Enemies.STALFOS, EnemyDistance.CLOSE, True, 2))),
+        (Locations.FOREST_TEMPLE_LOWER_STALFOS_POT1, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOWER_STALFOS_POT2, lambda bundle: can_break_pots(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_LOWER_STALFOS, world, [
+        (Regions.FOREST_TEMPLE_NORTH_CORRIDOR, lambda bundle: True)
+    ])
+    
+    ## Forest Temple NW Outdoors Lower
+    # Events
+    add_events(Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, world, [
+        (EventLocations.FOREST_TEMPLE_STICKS, Events.CAN_FARM_STICKS, lambda bundle: can_get_deku_baba_sticks(bundle)),
+        (EventLocations.FOREST_TEMPLE_NUTS, Events.CAN_FARM_NUTS, lambda bundle: can_get_deku_baba_nuts(bundle))
+    ])
+    # Locations
+    add_locations(Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, world, [
+        (Locations.FOREST_TEMPLE_WEST_COURTYARD_RIGHT_HEART, lambda bundle: (can_use(Items.BOOMERANG, bundle) and 
+                                                                             can_do_trick(Tricks.FOREST_OUTDOORS_HEARTS_BOOMERANG, bundle))),
+        (Locations.FOREST_TEMPLE_WEST_COURTYARD_LEFT_HEART, lambda bundle: (can_use(Items.BOOMERANG, bundle) and 
+                                                                            can_do_trick(Tricks.FOREST_OUTDOORS_HEARTS_BOOMERANG, bundle)))
```

Comment:
Sadly these 2 locations also need an unique region as they exist in 2 regions in Ship. Keep in mind both places where they exist have different rules to get these hearts, and don't give the new region exits.